cns/ipampool/monitor.go

```diff
@@ -52,9 +52,13 @@ type Monitor struct {
 	once        sync.Once
 }
 
-// Global Variables for Subnet, Subnet Address Space and Subnet ARM ID
+// Global Variables:
+// For Subnet, Subnet Address Space and Subnet ARM ID
 var subnet, subnetCIDR, subnetARMID string
 
+// For List of Unique Primary IP Addresses
+var ipAddresses map[string]int
```

Comment:
These definitely shouldn't be globals; they are not properties of the Monitor, but properties of the Pool (really, of the NCs). I think they belong in the state/metastate. 
There's a problem here in that these are per-NC and we can have multiple NCs, but I don't have a suggestion on how to handle that just yet and until we _do_ have multiple NCs we don't need to deal with it

Reply:
Localized this map.